src/main/frontend/sections/Toolbar.tsx

```diff
@@ -16,8 +16,52 @@ export const Toolbar: FC = () => {
 	const scriptContext = useContext(ScriptContext);
 	const script = scriptContext.draft;
 
+	const samples: Sample[] = fetchSamples();
 	const [, copy] = useCopyToClipboard();
 
+	function fetchSamples() {
```

Comment:
Since this uses the `useContext` hook, it becomes a hook as well. Either rename it to `useSamples` to signify that (according to React naming conventions) or move the 

```ts
const environmentContext = useContext(EnvironmentContext);
```

call outside (preferable) the function and pass in `environmentContext` (either implicitly as closed-over variable or explicitly as argument).